src/poetry/vcs/git/backend.py

```diff
@@ -331,16 +332,28 @@ def _clone_submodules(cls, repo: Repo) -> None:
         repo_root = Path(repo.path)
         modules_config = repo_root.joinpath(".gitmodules")
 
+        # A relative URL by definition starts with ../ or ./
+        relative_submodule_regex = re.compile("^(\\.\\./|\\./),?")
+
         if modules_config.exists():
             config = ConfigFile.from_path(str(modules_config))
 
             url: bytes
             path: bytes
             submodules = parse_submodules(config)
+
             for path, url, name in submodules:
                 path_relative = Path(path.decode("utf-8"))
                 path_absolute = repo_root.joinpath(path_relative)
 
+                url_string = url.decode("utf-8")
+                final_url = url_string
```

Comment:
If we're just going to clobber final_url anyway, it feels like we should just update url_string in place instead of introducing another variable.

Reply:
Resolved in latest commit.